### ISSUE TITLE
always have a blank line at first line without doctype in pretty mode.

### ISSUE THREAD

**Issue by dxcqcv** [pugjs/pug#2443]
I use pug to write jekyll make github page, so when I used yml front matter in pug, it always compiled a blank line at first line, so code like

    |---
    |layout: default
    |title: Shang Wen Long's blog
    |---
    .hello index contents

be compiled to 

    /******** Here is a blank line **********/
    ---
    layout: default
    title: Shang Wen Long's blog
    ---
    <div class="hello">index contents</div>

Through my test, in pretty mode because jekyll needs yml front matter format, there is blank line at first line without doctype, but with doctype everything is OK

so the first blank line is for doctype, I guess, but in this case I wanna remote it and keep in pretty mode, and to delete blank line every time is not fun, how to do it? or to fix this glitch. help, thx~

**Comment by ForbesLindesay** [comment]
We generally don't fix bugs in pretty printing mode.  You can force a blank line via `= '\n'`.  If it produces a blank first line with pretty mode disabled, that is a bug.

**Comment by TimothyGu** [comment]
Yeah that's a bug I tried to fix last year but couldn't do it. If all else fails, just `pug.render(...).trim()`.
